runtime/src/builtins.rs

```diff
@@ -213,48 +246,38 @@ fn dummy_process_instruction(
 /// Dynamic feature transitions for builtin programs
 fn builtin_feature_transitions() -> Vec<BuiltinFeatureTransition> {
     vec![
-        BuiltinFeatureTransition::Add {
-            builtin: Builtin::new(
-                "compute_budget_program",
-                solana_sdk::compute_budget::id(),
-                solana_compute_budget_program::process_instruction,
-            ),
-            feature_id: feature_set::add_compute_budget_program::id(),
-        },
-        BuiltinFeatureTransition::RemoveOrRetain {
-            previously_added_builtin: Builtin::new(
-                "secp256k1_program",
-                solana_sdk::secp256k1_program::id(),
-                dummy_process_instruction,
-            ),
-            addition_feature_id: feature_set::secp256k1_program_enabled::id(),
-            removal_feature_id: feature_set::prevent_calling_precompiles_as_programs::id(),
-        },
-        BuiltinFeatureTransition::RemoveOrRetain {
-            previously_added_builtin: Builtin::new(
-                "ed25519_program",
-                solana_sdk::ed25519_program::id(),
-                dummy_process_instruction,
-            ),
-            addition_feature_id: feature_set::ed25519_program_enabled::id(),
-            removal_feature_id: feature_set::prevent_calling_precompiles_as_programs::id(),
-        },
-        BuiltinFeatureTransition::Add {
-            builtin: Builtin::new(
-                "address_lookup_table_program",
-                solana_address_lookup_table_program::id(),
-                solana_address_lookup_table_program::processor::process_instruction,
-            ),
-            feature_id: feature_set::versioned_tx_message_enabled::id(),
-        },
-        BuiltinFeatureTransition::Add {
-            builtin: Builtin::new(
-                "zk_token_proof_program",
-                solana_zk_token_sdk::zk_token_proof_program::id(),
-                with_program_logging!(solana_zk_token_proof_program::process_instruction),
-            ),
-            feature_id: feature_set::zk_token_sdk_enabled::id(),
-        },
+        BuiltinFeatureTransition::new_add(
+            "compute_budget_program",
+            solana_sdk::compute_budget::id(),
+            solana_compute_budget_program::process_instruction,
+            feature_set::add_compute_budget_program::id(),
+        ),
+        BuiltinFeatureTransition::new_remove_or_retain(
+            "secp256k1_program",
+            solana_sdk::secp256k1_program::id(),
+            dummy_process_instruction,
+            feature_set::secp256k1_program_enabled::id(),
+            feature_set::prevent_calling_precompiles_as_programs::id(),
+        ),
```

Comment:
Please construct the enum variants directly here without the constructor methods. The constructor methods make it difficult to know what each feature arg is used for.

Reply:
hmm I was trying to avoid exposing implementation details.  Figured it's fine because all of the behavior is selfcontained

Reply:
I think my original version was fine

Reply:
Sigh. I can hold my nose for a Rust release cycle

Reply:
Ha, sorry man. I just don't trust that things won't get refactored and broken again before it's all removed.